File: app/src/main/java/bonoj/me/athenaeum/root/ApplicationComponent.java
package bonoj.me.athenaeum.root;

import javax.inject.Singleton;

import bonoj.me.athenaeum.books.BooksActivity;
import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {

    void inject(AthenaeumApplication application);

    void inject(BooksActivity booksActivity);

    //void inject(DetailsActivity detailsActivity);
}

File: app/src/main/java/bonoj/me/athenaeum/root/ApplicationComponent.kt
package bonoj.me.athenaeum.root

import bonoj.me.athenaeum.books.BooksActivity
import bonoj.me.athenaeum.details.DetailsActivity
import bonoj.me.athenaeum.details.DetailsViewModel
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    fun inject(application: AthenaeumApplication)

    fun inject(booksActivity: BooksActivity)

    fun inject(detailsActivity: DetailsActivity)

    fun inject(detailsViewModel: DetailsViewModel)
}
